### ISSUE TITLE
Add new React component for view (read only) rows in the editor

### ISSUE THREAD

**Issue by ta2-1** [translate/pootle#5037]
It's necessary to change layout for view rows for addressing #4976.
I think this is a good chance to get rid of `editor/units/xhr_view.html` template and create a new React component for rendering view (read only) units in the editor.

**Comment by julen** [comment]
I'm totally 👍 on converting such templates into components, but it's worth understanding the drawbacks and limitations when converting the view rows at the current state of things.

Unless what you are proposing is rendering individual components independently (via e.g. `ReactRenderer.render()` for every single row), from my understanding this would require something quite dirty by resourcing to [`dangerouslySetInnerHTML`](https://facebook.github.io/react/tips/dangerously-set-inner-html.html), which would inject an HTML fragment retrieved via JSON, and which will also contain certain components as of #4171:
```jsx
function getEditingWidget() {
  return {
    __html: editorAsHTMLString,
  };
}

return (
  <Editor>
    <ViewRow/>
    <ViewRow/>
    <EditRow dangerouslySetInnerHTML={getEditingWidget()}/>
    <ViewRow/>
    <ViewRow/>
  </Editor>
);
```

One would need to check, but I'm not even sure that would work under these circumstances.

--

Since these elements/components fall down under the same umbrella of the rendering tree/subtree, personally I'd favor a bottom-up approach, where components are gradually introduced from the leafs up towards parents. 

So in order to convert the view row code into a component, lower levels need to be componentized, which includes the entire editing row (_editor/units/edit.html_). As mentioned, part of this work is already done in #4171 (textareas).

**Comment by ta2-1** [comment]
Addressed in #5061
